File: src/integration/spring/src/jvmTest/java/community/flock/wirespec/integration/spring/java/application/Configuration.java
package community.flock.wirespec.integration.spring.java.application;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.web.reactive.function.client.WebClient;

@org.springframework.context.annotation.Configuration
public class Configuration {
    private final Logger log = LoggerFactory.getLogger(getClass());

    public Configuration() {
        log.info("Initializing application configuration");
    }

    @Bean("wirespecSpringWebClient")
    public WebClient webClientForWirespec(@Value("${server.port}") int port) {
        log.info("Creating custom webClient");
        return WebClient.create("http://localhost:" + port);
    }
}

File: src/integration/spring/src/jvmTest/kotlin/community/flock/wirespec/integration/spring/kotlin/application/Configuration.kt
package community.flock.wirespec.integration.spring.kotlin.application

import org.slf4j.LoggerFactory.getLogger
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
open class Configuration {
    private val log = getLogger(javaClass)

    init {
        log.info("Initializing application configuration")
    }

    @Bean("wirespecSpringWebClient")
    open fun webClientForWirespec(
        @Value("\${wirespec.spring.webclient.base-url}") baseUrl: String,
    ): WebClient {
        log.info("Creating custom webClient with base URL {}", baseUrl)
        return WebClient.create(baseUrl)
    }
}
